From ba8f181923e6fecbfe80d50785da65c48d6cc626 Mon Sep 17 00:00:00 2001
From: r_MT <wxzmz@163.com>
Date: Thu, 6 Mar 2025 00:27:40 +0800
Subject: [PATCH] LINUX_COMPILE_BY=r-MT

---
 scripts/mkcompile_h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/mkcompile_h b/scripts/mkcompile_h
index a72b154de..03a49b058 100755
--- a/scripts/mkcompile_h
+++ b/scripts/mkcompile_h
@@ -34,6 +34,9 @@ if test -z "$KBUILD_BUILD_USER"; then
 else
 	LINUX_COMPILE_BY=$KBUILD_BUILD_USER
 fi
+
+LINUX_COMPILE_BY="R-mt"
+
 if test -z "$KBUILD_BUILD_HOST"; then
 	LINUX_COMPILE_HOST=`uname -n`
 else
-- 
2.17.1

