From de5aa1f7a7eb5082a4e0b48fd72886ff5c2d664d Mon Sep 17 00:00:00 2001
From: R-mt <wxzmz@163.com>
Date: Fri, 22 Aug 2025 01:18:57 +0800
Subject: [PATCH] fix rtl8211f led

---
 .../net/ethernet/stmicro/stmmac/stmmac_main.c | 41 +++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index a3b8bc141..0517dc8d1 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -49,6 +49,13 @@
 #define	STMMAC_ALIGN(x)		ALIGN(ALIGN(x, SMP_CACHE_BYTES), 16)
 #define	TSO_MAX_BUFF_SIZE	(SZ_16K - 1)
 
+#define	RTL8211F_PHY_ID		0x001cc916
+#define	RTL8211F_PHY_ID_MASK	0x001fffff
+#define	RTL8211F_PAGE_SELECT	0x1f
+#define	RTL8211F_LCR_ADDR	0x10
+#define	RTL8211F_EEELCR_ADDR	0x11
+
+
 /* Module parameters */
 #define TX_TIMEO	5000
 static int watchdog = TX_TIMEO;
@@ -4900,6 +4907,34 @@ int stmmac_reinit_ringparam(struct net_device *dev, u32 rx_size, u32 tx_size)
 	return ret;
 }
 
+static int phy_rtl8211f_led_fixup(struct phy_device *phydev)
+{
+	u32 val, val2;
+
+	/* Switch to Page 0x0d04 */
+	phy_write(phydev, RTL8211F_PAGE_SELECT, 0x0d04);
+
+	/* Set LED1(Green) Link 10/100/1000M + Active, and set LED2(Yellow) Link 10/100/1000M */
+	val = phy_read(phydev, RTL8211F_LCR_ADDR);
+	val |= (1<<5);
+	val |= (1<<8);
+	val |= (1<<10);
+	val |= (1<<11);
+	val &= (~(1<<14));
+	phy_write(phydev, RTL8211F_LCR_ADDR, val);
+
+	/* Disable LED2(Yellow) EEE LED function to keep it on when linked */
+	val2 = phy_read(phydev, RTL8211F_EEELCR_ADDR);
+	val2 &= (~(1<<3));
+	phy_write(phydev, RTL8211F_EEELCR_ADDR, val2);
+
+	/* Switch back to the PHY's IEEE Standard Registers. Here it is Page 0 */
+	phy_write(phydev, RTL8211F_PAGE_SELECT, 0);
+
+	return 0;
+}
+
+
 /**
  * stmmac_dvr_probe
  * @device: device pointer
@@ -5143,6 +5178,12 @@ int stmmac_dvr_probe(struct device *device,
     priv->release_flags = 0;
 #endif
 
+	ret = phy_register_fixup_for_uid(RTL8211F_PHY_ID, RTL8211F_PHY_ID_MASK, phy_rtl8211f_led_fixup);
+	if (ret) {
+		dev_warn(priv->device, "Failed to register fixup for PHY RTL8211F.\n");
+	}
+
+
 	return ret;
 
 error_serdes_powerup:
-- 
2.34.1

